### ISSUE TITLE
Bidirectional Relationships

### ISSUE THREAD

**Issue by antoine-waag** [vinkla/extended-acf#157]
I would like to implement a bidirectional field with a Relationship component but it seems not implemented yet.

https://www.advancedcustomfields.com/resources/bidirectional-relationships/

**Comment by vinkla** [comment]
You're welcome to submit a pull request to add this feature. The challenge is that the keys are automatically generated. I haven't thoroughly examined this yet, but it seems that users would need to set custom field keys.

**Comment by antoine-waag** [comment]
@vinkla yes, the most difficult part is to get the field key reference from a field name and the post types list defined in the RelationShip field.

It seems, there is no function to get that key reference easily. All functions ask for `$post_id`

If you have any suggestion to achieve this issue, please let me know

**Comment by kun18911** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by vinkla** [comment]
@antoine-waag I wonder if we can simply add this to the configuration and allow users to [set custom field keys](https://github.com/vinkla/extended-acf#key) for fields they want to use in bidirectional relationships. What do you think? Would you be willing to make a pull request?

**Comment by kaelansmith** [comment]
I have achieved bidirectional relationships in the past by doing this:

```php
// This field is attached to an "Awards" post type:
Relationship::make('Related Project')
       ->postTypes(['project'])
       ->key('field_award_related_project') // manually specify a field key in order to reference it via bidirectional relationship field
       ->withSettings([
            'bidirectional' => 1,
            'bidirectional_target' => ['field_project_related_award']
       ])

// This field is attached to a "Project" post type:
Relationship::make('Related Award')
       ->postTypes(['award'])
       ->key('field_project_related_award') // manually specify a field key in order to reference it via bidirectional relationship field
       ->withSettings([
            'bidirectional' => 1,
            'bidirectional_target' => ['field_award_related_project']
       ]),
```

It would be nice if there was a built-in convenience method such as `bidirectionalTarget` which received a field key and adds `['bidirectional' => 1, 'bidirectional_target' => [$key]]` to the field's settings. It might also be nice to have a solution that doesn't require manually setting keys, to avoid human error, but probably not necessary?

**Comment by vinkla** [comment]
Interesting, @kaelansmith! Adding a method `bidirectional()` would be relatively easy to implement. However, automatically setting keys is more challenging. This situation differs from conditional logic, which progresses through the field tree. Perhaps enforcing the use of custom keys is the best approach. Would you be willing to create a pull request to add the `bidirectional` method?

**Comment by kaelansmith** [comment]
@vinkla done! #161

**Comment by vinkla** [comment]
This has been added in #161 and can be tested using the branch alias 14.3. Please test it locally if possible.
